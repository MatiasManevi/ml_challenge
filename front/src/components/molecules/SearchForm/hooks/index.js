export { default as useSearchForm } from 'components/molecules/SearchForm/hooks/useSearchForm';
