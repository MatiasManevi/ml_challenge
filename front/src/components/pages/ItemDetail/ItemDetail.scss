@import 'assets/scss/variables';

.item-detail {
	background-color: $color-white;
	padding: $space-32;
}

.item-detail__picture {
	width: $detail-img-width;
}

.item-detail__description-title {
	font-size: $font-size-28;
	font-weight: 100;
	margin-bottom: $space-32;
}

.item-detail__description-content {
	font-size: $font-size-16;
	color: $color-neutral;
}

.item-detail__condition {
	font-size: $font-size-14;
	color: $color-neutral-darker;
	margin-bottom: $space-16;
}

.item-detail__name {
	font-size: $font-size-24;
	font-weight: bold;
	margin-bottom: $space-32;
}
