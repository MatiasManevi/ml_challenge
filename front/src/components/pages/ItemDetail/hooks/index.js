export { default as useItemDetail } from 'components/pages/ItemDetail/hooks/useItemDetail';
