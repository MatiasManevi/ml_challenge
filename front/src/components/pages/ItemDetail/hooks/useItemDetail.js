import React from 'react';
import axios from 'axios';

import { API_BASE_URL } from 'utils';

const useItemDetail = (id) => {
	const [itemDetail, setItemDetail] = React.useState(null);

	React.useEffect(async () => {
		try {
			const response = await axios.get(`${API_BASE_URL}/items/${id}`);
			setItemDetail(response.data.item);
		} catch (e) {
			console.error(e);
		}
	}, []);

	return itemDetail;
};

export default useItemDetail;
