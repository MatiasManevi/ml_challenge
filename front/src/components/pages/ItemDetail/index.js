export { default } from 'components/pages/ItemDetail/ItemDetail';
