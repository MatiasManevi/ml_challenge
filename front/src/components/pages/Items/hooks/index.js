export { default as useItems } from 'components/pages/Items/hooks/useItems';
