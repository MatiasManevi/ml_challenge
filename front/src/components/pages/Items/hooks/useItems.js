import React from 'react';

import { paginationQuery } from 'utils';
import ItemService from 'services/items';

const useItems = () => {
	const [items, setItems] = React.useState([]);
	const [categories, setCategories] = React.useState([]);
	const search = new URLSearchParams(window.location.search).get('search');

	React.useEffect(async () => {
		try {
			const response = await ItemService.get(search + paginationQuery);
			setItems(response.items);
			setCategories(response.categories);
		} catch (e) {
			console.error(e);
		}
	}, [search]);

	return {
		items,
		categories
	};
};

export default useItems;
