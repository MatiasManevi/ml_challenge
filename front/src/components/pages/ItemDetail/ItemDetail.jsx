import React from 'react';
import Grid from '@material-ui/core/Grid';

import './ItemDetail.scss';
import { Price, PrimaryButton } from 'components/atoms';
import { BreadcrumbMenu } from 'components/organisms';
import { Content } from 'components/templates';
import { useItemDetail } from './hooks';

const ItemDetail = ({ match }) => {
	const itemDetail = useItemDetail(match.params.id);

	if (!itemDetail) {
		return null;
	}

	const {
		picture,
		description,
		condition,
		sold_quantity,
		title,
		price,
		free_shipping
	} = itemDetail;

	return (
		<Content>
			<BreadcrumbMenu />
			<Grid container className="item-detail">
				<Grid item xs={9}>
					<img
						src={picture}
						alt={`imagen de ${title}`}
						className="item-detail__picture"
					/>
					<h3 className="item-detail__description-title">Descripci√≥n del producto</h3>
					<p className="item-detail__description-content">{description}</p>
				</Grid>

				<Grid item xs={10} sm={3}>
					<p className="item-detail__condition">
						{condition} - {sold_quantity} vendidos
					</p>
					<h1 className="item-detail__name">{title}</h1>
					<Price price={price} freeShipping={free_shipping} small={false} />
					<PrimaryButton text="Comprar" clickHandler={() => console.log('compra')} />
				</Grid>
			</Grid>
		</Content>
	);
};

export default ItemDetail;
