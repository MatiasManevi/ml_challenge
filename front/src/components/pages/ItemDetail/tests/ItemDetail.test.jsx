import React from 'react';
import { shallow } from 'enzyme';

import ItemDetail from '../ItemDetail';
import * as hooks from '../hooks';
import item from './item.fixtures';

jest.mock('../hooks');

const stateSpy = jest.spyOn(hooks, 'useItemDetail');

stateSpy.mockReturnValue(item);

describe('ItemDetail', () => {
	it('should match snapshot', () => {
		const props = { match: { params: { id: item.id } } };
		const wrapper = shallow(<ItemDetail {...props} />);
		expect(wrapper).toMatchSnapshot();
	});
});
