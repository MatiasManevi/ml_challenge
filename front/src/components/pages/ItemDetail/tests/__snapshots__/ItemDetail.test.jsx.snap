// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDetail should match snapshot 1`] = `
<Content>
  <BreadcrumbMenu />
  <WithStyles(ForwardRef(Grid))
    className="item-detail"
    container={true}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={9}
    >
      <img
        alt="imagen de Guantes Termicos Primera Piel Ski Trekking Running Softshell"
        className="item-detail__picture"
        src="http://http2.mlstatic.com/D_970636-MLA45626185241_042021-O.jpg"
      />
      <h3
        className="item-detail__description-title"
      >
        Descripción del producto
      </h3>
      <p
        className="item-detail__description-content"
      >
        Ale1416
Tienda de Deportes – ProShop

CARACTERISTICAS PRINCIPALES:

. Tela Softshell
. Retencion de la temperatura corporal
. Absorben el calor y eliminan la humedad
. Totalmente elasticos (se ajustan a cualquier mano)
. Suaves al tacto
. Talles Unicos - Unisex

DIAS Y HORARIOS DE ATENCION
De Lunes a Viernes de 8:00 Hs A 17:00 Hs
Sabado, Domingo y Feriados cerrado


ESTAMOS POR CABALLITO 
Cerca de Av. Directorio y Av. Jose Maria Moreno

Si venis en colectivo
Te acercan mas de 20 lineas

Si venis en subte
Est. Jose Maria Moreno - Linea E

Si venis en tren
Est. Caballito – Linea Sarmiento



FORMAS DE ENVIO Y PLAZOS DE ENTREGA

Correo (MercadoEnvio)
El despacho se realiza por correo con seguimiento online. El plazo de entrega y costo del mismo lo calculas poniendo tu codigo postal en el zocalo que esta debajo del precio.

Moto Mensajeria (CABA – GBA)
El envio es en el dia y el costo varia segun el destino. Los dias de lluvia tienen un recargo del 50% y tambien se cobra recargo extra por bulto de gran tamaño.

Micro
Es para los productos que no aceptan MercadoEnvios y para comprar varios articulos pagando un solo envio. El destino es la terminal o las oficinas de la empresa en tu localidad y la demora es de 48 hs habiles. Todos los paquetes tienen un adicional de $50 para cubrir los gastos de embalaje y traslado al lugar de despacho
      </p>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      sm={3}
      xs={10}
    >
      <p
        className="item-detail__condition"
      >
        Nuevo
         - 
        500
         vendidos
      </p>
      <h1
        className="item-detail__name"
      >
        Guantes Termicos Primera Piel Ski Trekking Running Softshell
      </h1>
      <Price
        freeShipping={false}
        price={
          Object {
            "amount": "439",
            "currency": "$",
            "decimals": 99,
          }
        }
        small={false}
      />
      <PrimaryButton
        clickHandler={[Function]}
        text="Comprar"
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</Content>
`;
